src/coreclr/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComponentActivator.cs

```diff
@@ -144,61 +144,69 @@ private static IsolatedComponentLoadContext GetIsolatedComponentLoadContext(stri
                                                          string methodName,
                                                          IntPtr delegateTypeNative)
         {
-            // Create a resolver callback for types.
-            Func<AssemblyName, Assembly> resolver = name => alc.LoadFromAssemblyName(name);
-
-            // Determine the signature of the type. There are 3 possibilities:
-            //  * No delegate type was supplied - use the default (i.e. ComponentEntryPoint).
-            //  * A sentinel value was supplied - the function is marked UnmanagedCallersOnly. This means
-            //      a function pointer can be returned without creating a delegate.
-            //  * A delegate type was supplied - Load the type and create a delegate for that method.
-            Type? delegateType;
-            if (delegateTypeNative == IntPtr.Zero)
+            if (Marshal.IsComSupported)
```

Comment:
Is there some kind of code style guidelines that stop you from doing this instead?

```
if (!Marshal.IsComSupported)
{
    throw new NotSupportException(SR.NotSupported_COM);
}
```

Reply:
I agree. The way these changes were made makes this seem much more impactful than it is. Indenting so much of noisy and there is always the surprise at the end of oh this just throws. The short branch should almost ways be first in my opinion.